noteorganiser/frames.py

```diff
@@ -837,6 +844,23 @@ def resetSize(self):
         self.font.setPointSize(self.defaultFontSize)
         self.text.setFont(self.font)
 
+    def setupAutoRefresh(self, source):
+        """add current file to QFileSystemWatcher and refresh when needed"""
+        # only setup if wanted
+        if self.info.refreshEditor:
+            self.fileSystemWatcher.addPath(source)
+            self.fileSystemWatcher.fileChanged.connect(
+                self.getAutoRefreshSignal)
+            self.log.info("added file %s to FileSystemWatcher" % source)
+
+    def getAutoRefreshSignal(self):
+        """refresh editor when needed"""
+        # wait some time for the change to finish
+        if not self.text.document().isModified():
```

Comment:
`self.text.document()` is the notebook in the editor. so if you modify it in noteorganiser, `isModified()` is `true`
we don't want to loose changes made in the editor, so only reload from file if it's **not** modified